src/Microsoft.TestPlatform.Common/ExtensionFramework/DataCollectionExtensionManager.cs

```diff
@@ -75,18 +73,22 @@ public static DataCollectorExtensionManager Create(IMessageLogger messageLogger)
     /// <returns>
     /// The DataCollectorExtensionManager.
     /// </returns>
-    public static DataCollectorExtensionManager Create(string extensionAssemblyFilePath, bool skipCache, IMessageLogger messageLogger)
+    public static DataCollectorExtensionManager Create(string extensionAssemblyFilePath, bool skipCache, IMessageLogger messageLogger,
+        TestPluginCache? testPluginCache = null)
     {
         TestPluginManager.GetTestExtensions<DataCollectorConfig, ObjectModel.DataCollection.DataCollector, IDataCollectorCapabilities, DataCollectorMetadata>(
             extensionAssemblyFilePath,
             out var unfilteredTestExtensions,
             out var filteredTestExtensions,
-            skipCache);
+            skipCache,
+            testPluginCache);
 
         return new DataCollectorExtensionManager(unfilteredTestExtensions, filteredTestExtensions, messageLogger);
     }
 }
 
+#nullable disable
```

Comment:
Moved this down so that the internal functions can check nulls while not having to change public api.